dotnet/src/Extensions/Planning.StepwisePlanner/StepwisePlanner.cs

```diff
@@ -160,7 +162,7 @@ public Plan CreatePlan(string goal)
                             nextStep.Observation = result;
                         }
                     }
-                    catch (Exception ex) when (!ex.IsCriticalException())
+                    catch (Exception ex)
                     {
                         nextStep.Observation = $"Error invoking action {nextStep.Action} : {ex.Message}";
                         this._logger?.LogWarning(ex, "Error invoking action {Action}", nextStep.Action);
```

Comment:
The exception is being swallowed and no log data is being stored about it.

LogWarning should be LogError or at least include the exception details.

Reply:
I will tag @lemillermicrosoft for input here.

> The exception is being swallowed

As far as I know, this is expected behavior for StepwisePlanner, when it receives an exception, it will put details in `Observation` field and proceed to the next step.

> LogWarning should be LogError

In case this is correct behavior, we shouldn't log it as error, because it doesn't break the application flow, instead it's just transient error. Previously we used Debug level here, and I updated it to Warning, and it's still not clear whether it was right choice.

> or at least include the exception details.

As far as I can see, we include `ex` instance as first parameter to `LogWarning` method.

Reply:
Yes thank you @dmytrostruk . I think LogDebug is probably best in this scenario (I was gonna change in another PR).